WordPress/gutenberg
# DataViews: make the move left/right controls in table header always available

## What?

Introduce a `enableMoving` that controls whether the "move to left"/"move to right" controls are visible in the table layout:

<img width="255" alt="Captura de ecrã 2024-08-20, às 16 28 23" src="https://github.com/user-attachments/assets/58964ec2-22f1-4f33-b354-a8e755f7fb21">

## Why?

Without this setting, the visibility of those controls depends on the other conditions:

- enableHiding: if the "hide column" option is displayed. `true` by default.
- enableSorting: if the sort options are displayed. `true` by default.
- If the field has a filter defined, it'll display the "add filter".

There are situations where the field can be hidden, it's not sortable, and doesn't have a filter, but we still want to display the move left/right controls.

## How?

Introduce a new `enableMoving` setting to the fields API. It's true by default.

## Testing Instructions

Nothing changes in terms of UI.

For testing the feature, go to the post-fields configuration and set the `title` field to `enableSorting: false`. Reload and verify that the table header controls are still displayed:

<img width="1207" alt="Captura de ecrã 2024-08-20, às 17 15 42" src="https://github.com/user-attachments/assets/815a1a19-7a26-4d61-a2e6-5ab02a3a3b7c">
